pep-0707.rst

```diff
@@ -0,0 +1,358 @@
+             }
+        }
+        return true;
+    }
+
+
+Note that any performance cost of the introspection can be mitigated via
+specialization, so it won't be a problem if we need to make it more
+sophisticated than this for some reason [:pep:`654`].
+
+
+Backwards Compatibility
+=======================
+
+All context managers that previously worked, will continue to work in the
+same way because the interpreter will call them with three args whenever
+they can accept three args. There may be context managers that previously
+did not work because their ``exit_func`` expected one argument. The call
+to ``__exit__`` would have caused a ``TypeError`` exception to be raised,
+and now the call would succeed. This could theoretically change the
+behaviour of existing code, but it is unlikely to be a problem in practice.
+
+The backwards compatibility concerns will show up in some cases when libraries
+try to migrate their context managers from the multi-arg to the single-arg
+signature. If ``__exit__`` or ``__aexit__`` is called by any code other than
+the interpreter's eval loop, the introspection does not automatically happen.
+For example, this will occur where a context manager is subclassed and its
+``__exit__`` method is called directly from the derived ``__exit__``. Such
+context managers will need to migrate to the single-arg version with their
+users, and may choose to offer a parallel API rather than breaking the
+existing one. Alternatively, a superclass can stay with the signature
+``__exit__(self, *args)``, and support both one and three args.
+
+
+How to Teach This
+=================
+
+The language tutorial will present the single-arg version, and the documentation
+for context managers will include a section on the legacy signatures of
+``__exit__`` and ``__aexit__``.
+
+
+Reference Implementation
+========================
+
+`CPython PR #101995 <https://github.com/python/cpython/pull/101995>`_
+implements the proposal of this PEP.
+
+
+Rejected Ideas
+==============
+
+Support ``__leave__(self, exc)``
+----------------------------------
+
+It was considered to support a method by a new name, such as ``__leave__``,
+with the new signature. This basically makes the programmer explicitly declare
+which signature they are intending to use, and avoid the need for introspection.
+I don't think this solution has benefits that justify the addition of a new
+dunder name.
+
+Support ``__leave__(self, exc)``, with trampolines
+----------------------------------------------------
+
+An enhancement of the previous idea that seems to promise a smoother migration
+was to add ``__leave__``, and make sure that any type that defines ``__leave__``
+or ``__exit__`` is augmented with a builtin version of the other. The builtin
+``__leave__`` calls ``__exit__``, and the builtin ``__exit__`` calls ``__leave__``.
+
+This became very complicated when it came to making it work for direct calls to
+a superclass::
+
+.. code-block:: python
```

Comment:
Either use the `::` shortcut or the explicit `.. code-block:: python`, not both:

https://pep-previews--3018.org.readthedocs.build/pep-0707/#support-leave-self-exc-with-trampolines

```suggestion
a superclass::
```